### DISCUSSION TITLE
Issue with stale data on npx servers.

### DISCUSSION THREAD

**Discussion by Pash10g** [modelcontextprotocol/.github#98] [Q&A]
### Pre-submission Checklist

- [X] I have checked that this question would not be more appropriate as an issue in a specific repository
- [X] I have searched existing discussions and documentation for answers

### Question Category

- [ ] Protocol Specification
- [ ] SDK Usage
- [X] Server Implementation
- [ ] General Implementation
- [ ] Documentation
- [X] Other

### Your Question

**Describe the bug**
When I use npx based servers with my npm package, I have MONGODB_URI env configuration in claude config, 
```
"mongodb": {
      "command": "npx",
      "args": [
        "-y" ,
        "@pash1986/mcp-server-mongodb"
      ],
     "env" : {
	"MONGODB_URI" : "mongodb+srv://<yourcluster>"
	}
    }
```

The data that was placed and sampled by Claude Desktop in the initial interaction with  "@pash1986/mcp-server-mongodb" npm remains constatn.

I replace the value to another MONGODB_URI instance... But it still goes into the first one I ever connected. What odd is after my Mac is getting restarted it still goes to the first ever specified MongoDB URI...

Is that expected in any way... I don't even know where to debug the clear and reload of env variables within the Claude Desktop processes....

An important note is that if new collections are added to original URI database , claude will read them.

**To Reproduce**
1. Use the https://www.npmjs.com/package/@pash1986/mcp-server-mongodb package. Point it to a MongoDB instance.
2.  Run a few interactions
3. Try to replace the URI to something else. 

**Expected behavior**
Changing the environment vars will result in Claude Desktop working against the current instance.

**Logs**
If applicable, add logs to help explain your problem.

**Additional context**

**Comment by Pash10g** [comment]
Anyone?

**Reply by slimslenderslacks** [reply]
@Pash10g when you're talking about stale data, are you talking about mcp resources or data coming back from tool calls? I've been wondering about something similar and so I looked at your uri scheme ( uri: `mcp-mongodb://${collection.name}/schema` ) only including collection names. Claude might cache resources with the same uri (even if they're from different databases). I guess I'm mostly wondering whether you see a new set of ReadResourceRequests after switching the MONGODB_URI.

**Comment by hichana** [comment]
I've experienced this developing my server. In my case I publish an NPM package, then was expecting the next time I open Claude that it'd detect there are new tools, but it doesn't appear to do so, nor does Cline which I've also been testing on. On the Tools MCP documentation I don't see anything about subscribing or publishing so I'm unclear how this happens in practice.

**Reply by jspahrsummers** [reply]
If you're pulling in your server via npm, it's likely that the npm package itself is being cached. You can use `npx --cache=/tmp/emptycache` to test whether bypassing existing caching resolves the issue.
